great_expectations/core/expectation_configuration.py

```diff
@@ -93,13 +105,19 @@ def make_expectation_context(self, data, **kwargs):
         return ExpectationContext(**data)
 
 
+class KWargDetailsDict(TypedDict):
+    domain_kwargs: tuple[str, ...]
+    success_kwargs: tuple[str, ...]
```

Comment:
AFIAK these are never meant to be mutated, so by making them `tuples` instead of lists we assure they are not modified and reduce the size of this `kwarg_lookup_dict` by about `15%` `61016b -> 51496b`.

Reply:
A benefit of using this type-dict is we don't have to deal with the union of all possible values for every key.

If this was just a dict we would have to narrow the union of all possible types for every field.
But because of the `TypeDict` we know that `domain_kwargs` & `success_kwargs` are always tuples, and `default_kwarg_values` are always a dict.